frontend/src/components/AvailabilityDropDown.tsx

```diff
@@ -0,0 +1,95 @@
+import { useEffect, useState } from "react";
+import styled from "styled-components";
+
+import {
+  ApplyButton,
+  DropDownPopup,
+  Dropdown,
+  DropdownIcon,
+  DropdownRow,
+  FilterRow,
+  FilterSubContainer,
+  FilterText,
+  Sort,
+} from "@/components/FilterCommon";
+
+const AvailabilityRow = styled(FilterRow)``;
+
+const FilterRadioButton = styled.img`
+  height: 20px;
+  width: 20px;
+`;
+
+const PopupBodyText = styled(Sort)`
+  font-weight: 400;
+  font-size: 12px;
+  margin-left: 0;
+`;
+
+export type AvailabilityDropDownProps = {
+  onApply(selected: number): void;
+  registerResetCallback(callback: () => void): void;
+};
+
+export const AvailabilityDropDown = (props: AvailabilityDropDownProps) => {
+  const [isActive, setIsActive] = useState(false);
+  const [selectedIdx, setSelectedIdx] = useState(0);
+  const [dropdownText, setDropdownText] = useState("Availabile");
+
+  const availabilityOptions: string[] = ["Available", "Leased"];
+
+  const resetFilter = () => {
+    setSelectedIdx(0);
+    setDropdownText("Availablity");
+  };
+
+  // useEffect(() => {
+  //   props.registerResetCallback(resetFilter);
+  // }, []);
```

Comment:
When running locally, I ran into a really weird bug where logging in would freeze the whole page (almost like there was an infinite loop somewhere). I _think_ I track it down to this `useEffect`. I commented this out in the two other dropdown components too and that fixed the log in bug 🤷🏼 . I'm still not entirely sure why that is happening. If you uncomment these lines, you should be able to replicate the bug.